### PR TITLE
Support unlisted public posts

### PR DESCRIPTION
**Pull request by annando**
This is the support for unlisted public posts in Friendica. We also transport this information to ActivityPub, so that systems like Mastodon won't publish these posts on their global feed as well.

I hope I haven't missed a place, but I will keep testing it for a few days (and I have to fix the conflicts), so this is currently a draft.

### CODE REVIEW

File: database.sql [hunk 1]
```diff
@@ -568,7 +568,7 @@ CREATE TABLE IF NOT EXISTS `item` (
 	`extid` varchar(255) NOT NULL DEFAULT '' COMMENT '',
 	`post-type` tinyint unsigned NOT NULL DEFAULT 0 COMMENT 'Post type (personal note, bookmark, ...)',
 	`global` boolean NOT NULL DEFAULT '0' COMMENT '',
-	`private` boolean NOT NULL DEFAULT '0' COMMENT 'distribution is restricted',
+	`private` tinyint unsigned NOT NULL DEFAULT 0 COMMENT 'distribution is restricted',
```
Comment (MrPetovan) [inline database.sql]: Please update the comment with the meaning of the different values.

File: database.sql [hunk 2]
```diff
@@ -1179,7 +1178,7 @@ CREATE TABLE IF NOT EXISTS `thread` (
 	`received` datetime NOT NULL DEFAULT '0001-01-01 00:00:00' COMMENT '',
 	`changed` datetime NOT NULL DEFAULT '0001-01-01 00:00:00' COMMENT '',
 	`wall` boolean NOT NULL DEFAULT '0' COMMENT '',
-	`private` boolean NOT NULL DEFAULT '0' COMMENT '',
+	`private` tinyint unsigned NOT NULL DEFAULT 0 COMMENT '',
```
Comment (MrPetovan) [inline database.sql]: Please update the comment with the meaning of the different values.

**Comment by annando** [comment]
With DFRN we don't transmit the value "2" in the "private" field to maintain compatibility. And in AP it doesn't matter for Friendica <-> Friendica.

**Comment by MrPetovan** [comment]
What does an unlisted post privacy ends up being on an unsupported Friendica node? Public or Private?

**Comment by annando** [comment]
"unlisted" is a public post, so "unlisted" is "public" on DFRN. Everything else wouldn't work.

**Comment by MrPetovan** [comment]
Ok, I just wanted to make sure. This probably should be said when selecting the unlisted choice that not all remote servers will honor it.

**Comment by annando** [comment]
I did several tests and everything appeared to work out fine. So I now declare this PR as ready to be merged.

File: mod/settings.php [hunk 3]
```diff
@@ -837,6 +839,10 @@ function settings_content(App $a)
 		'$field' => ['hidewall', DI::l10n()->t('Hide your profile details from anonymous viewers?'), $a->user['hidewall'], DI::l10n()->t('Anonymous visitors will only see your profile picture, your display name and the nickname you are using on your profile page. Your public posts and replies will still be accessible by other means.')],
 	]);
 
+	$unlisted = Renderer::replaceMacros($opt_tpl, [
+		'$field' => ['unlisted', DI::l10n()->t('Should public posts be unlisted?'), DI::pConfig()->get(local_user(), 'system', 'unlisted'), DI::l10n()->t('Your public posts will not appear on the community page or in search results, nor will they be transported to relay servers. However they can appear on public feeds from other servers.')],
```
Comment (MrPetovan) [inline mod/settings.php]:
```suggestion
		'$field' => ['unlisted', DI::l10n()->t('Make public posts unlisted'), DI::pConfig()->get(local_user(), 'system', 'unlisted'), DI::l10n()->t('Your public posts will not appear on the community pages or in search results, nor be sent to relay servers. However they can still appear on public feeds on remote servers.')],
```

**Pull request merged**